srml/support/src/storage/generator.rs

```diff
@@ -119,7 +119,7 @@ pub trait StorageValue<T: codec::Codec> {
 	}
 
 	/// Mutate this value
-	fn mutate<F: FnOnce(&mut Self::Query), S: Storage>(f: F, storage: &S);
+	fn mutate<R, F: FnOnce(&mut Self::Query) -> R, S: Storage>(f: F, storage: &S) -> R;
```

Comment:
I think that I would change the closure to take the value by value and return the mutated value directly. 
If we really need a version, where the return value is not the mutated value, we should introduce a new function later.

This would simplify:
```
		let result = <Foo<T>>::mutate(|foo| {
			*foo = *foo + increase_by;
			*foo
		});
```

to

```
		let result = <Foo<T>>::mutate(|foo| { foo + increase_by });
```

Reply:
That was my second question, but from @gavofyork 's reply, I think Gav wants a more flexible way to return ANY thing from the internal closure.

Reply:
Yeah, I realized that after writing my comment. Sorry.